api/services/post.js

```diff
@@ -0,0 +1,240 @@
+'use strict';
+
+var self = post;
+module.exports = self;
+
+var async = require('async');
+var path = require('path');
+var _ = require('underscore');
+var spawn = require('child_process').spawn;
+var fs = require('fs');
+
+var configHandler = require('../../common/configHandler.js');
+
+var apiConfig = require('./apiConfig.js');
+
+function post(req, res) {
+  var bag = {
+    reqBody: req.body,
+    resBody: {},
+    params: {},
+    tmpScript: '/tmp/service.sh'
+  };
+
+  bag.who = util.format('services|%s', self.name);
+  logger.info(bag.who, 'Starting');
+
+  async.series([
+      _checkInputParams.bind(null, bag),
+      _getServiceConfig.bind(null, bag),
+      _generateServiceConfig.bind(null, bag),
+      _generateInitializeEnvs.bind(null, bag),
+      _generateScript.bind(null, bag),
+      _writeScriptToFile.bind(null, bag),
+      _bootService.bind(null, bag),
+      _post.bind(null, bag),
+    ],
+    function (err) {
+      logger.info(bag.who, 'Completed');
+      if (err)
+        return respondWithError(res, err);
+
+      sendJSONResponse(res, bag.resBody);
+    }
+  );
+}
+
+function _checkInputParams(bag, next) {
+  var who = bag.who + '|' + _checkInputParams.name;
+  logger.verbose(who, 'Inside');
+
+  if (!bag.reqBody)
+    return next(
+      new ActErr(who, ActErr.BodyNotFound, 'Missing body')
+    );
+
+  if (_.isEmpty(bag.reqBody.name))
+    return next(
+      new ActErr(who, ActErr.ParamNotFound,
+        'Data not found :name')
+    );
+  bag.name = bag.reqBody.name;
+
+  if (_.isEmpty(bag.reqBody.replicas))
+    return next(
+      new ActErr(who, ActErr.ParamNotFound,
+        'Data not found :replicas')
+    );
+
+  bag.replicas = bag.reqBody.replicas;
+
+  return next();
+}
+
+function _getServiceConfig(bag, next) {
+  var who = bag.who + '|' + _generateServiceConfig.name;
+  logger.verbose(who, 'Inside');
+
+  configHandler.get('services',
+    function (err, config) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.OperationFailed, err)
+        );
+
+      if (_.isEmpty(config))
+        return next(
+          new ActErr(who, ActErr.DataNotFound, 'No service configs present'));
+
+      if (!_.has(config, bag.name))
+        return next(
+          new ActErr(who, ActErr.DataNotFound, 'No service configs present ' +
+          ' for :' + bag.name)
+        );
+
+      bag.services = config;
+      bag.serviceConfig = config[bag.name];
+      return next();
+    }
+  );
+}
+
+function _generateServiceConfig(bag, next) {
+  var who = bag.who + '|' + _generateServiceConfig.name;
+  logger.verbose(who, 'Inside');
+
+  var configGenerator = null;
+  if (bag.name === 'api') configGenerator = apiConfig;
```

Comment:
for all the microservices, the name will map to same handler since they all have same configs(env, mounts etc). so this should work for now. if the logic gets complicated, we can add a map in next heap issue